upf/upf_cli.c

```diff
@@ -635,6 +641,85 @@ VLIB_CLI_COMMAND (upf_show_spec_release_command, static) =
 };
 /* *INDENT-ON* */
 
+static clib_error_t *
+upf_node_id_command_fn (vlib_main_t * vm,
+			unformat_input_t * main_input,
+			vlib_cli_command_t * cmd)
+{
+  upf_main_t *gtm = &upf_main;
+  unformat_input_t _line_input, *line_input = &_line_input;
+  clib_error_t *error = NULL;
+  ip46_address_t ip46;
+  u8 *fqdn = 0;
+  pfcp_node_id_t *node_id = &gtm->node_id;
+
+  if (node_id->type == NID_FQDN)
+    vec_free (node_id->fqdn);
+
+  if (!unformat_user (main_input, unformat_line_input, line_input))
+    return 0;
+
+  while (unformat_check_input (line_input) != UNFORMAT_END_OF_INPUT)
+    {
+      if (unformat (line_input, "fqdn %_%v%_", &fqdn))
+	{
+	  node_id->type = NID_FQDN;
+	  node_id->fqdn = upf_name_to_labels (fqdn);
+	  vec_free (fqdn);
+	}
+      else if (unformat (line_input, "ip4 %U",
+			 unformat_ip46_address, &ip46, IP46_TYPE_ANY))
+	{
+	  node_id->type = NID_IPv4;
+	  node_id->ip = ip46;
+	}
+      else if (unformat (line_input, "ip6 %U",
+			 unformat_ip46_address, &ip46, IP46_TYPE_ANY))
+	{
+	  node_id->type = NID_IPv6;
+	  node_id->ip = ip46;
+	}
+      else
+	{
+	  error = unformat_parse_error (line_input);
+	  return 0;
+	}
+    }
+
+  return NULL;
+
+}
+
+/* *INDENT-OFF* */
+VLIB_CLI_COMMAND (upf_node_id_command, static) = {
+    .path = "upf node-id",
+    .short_help = "upf node-id <type:fqdn/ip4/ip6> <value>]",
```

Comment:
for VPP that is a very unusual format specification
```suggestion
    .short_help = "upf node-id ( fqdn <fqdn> | ip4 <ip4-addr> | ip6 <ip6-addr> )",
```

Reply:
Done